lua/weapons/weapon_laserpointer.lua

```diff
@@ -0,0 +1,326 @@
+-- This file is subject to copyright - contact swampservers@gmail.com for more information.
+
+SWEP.PrintName = "Laser Pointer"
+SWEP.Author = "Britain1944"
+SWEP.Instructions = "Left click: Turn on and off the laser.\nRight click: Change laser power\nRepurchase for more ammo."
+
+SWEP.Category = "Other"
+
+SWEP.Spawnable = true
+SWEP.Slot = 3
+SWEP.SlotPos = 0
+SWEP.DrawAmmo = false
+SWEP.DrawCrosshair = true
+
+game.AddAmmoType({
+	name = "Battery",
+	dmgtype = nil,
+	tracer = nil,
+	plydmg = 0,
+	npcdmg = 0,
+	force = 0,
+	minsplash = 0,
+	maxsplash = 0
+})
+
+SWEP.Primary.Ammo = "Battery"
+SWEP.Primary.ClipSize = 100
+SWEP.Primary.DefualtClip = 100
+SWEP.Primary.Automatic = true
+
+SWEP.Secondary.Ammo = "none"
+SWEP.Secondary.ClipSize = -1
+SWEP.Secondary.DefaultClip = -1
+SWEP.Secondary.Automatic = false
+
+SWEP.ViewModel = "models/brian/laserpointer.mdl"
+SWEP.WorldModel = "models/brian/laserpointer.mdl"
+
+SWEP.laserorgpos = Vector(0,0,0)
+SWEP.laserorgang = Angle(0,0,0)
+SWEP.LaserBeamOn = false
+SWEP.LaserPower = 0
+SWEP.LaserColors = {
+	Color(255,0,0,255),
+	Color(0,255,0,255),
+	Color(255,0,255,255)
+}
+
+if CLIENT then
+	hook.Add("HUDPaint", "BatteryBar", function()
+		local wep = LocalPlayer():GetActiveWeapon()
+		if IsValid(wep) and wep:GetClass() == "weapon_laserpointer" then
+			draw.RoundedBox(8, (ScrW()/2)-64, ScrH()-48, 128, 32, Color(0, 0, 0, 128))
+			draw.RoundedBox(0, (ScrW()/2)-58, ScrH()-42, 1.16*wep:Clip1(), 20, wep.LaserColors[wep.LaserPower+1])
+			draw.DrawText("Battery", "DermaLarge", (ScrW()/2), ScrH()-84, Color(255, 255, 255, 255), TEXT_ALIGN_CENTER)
+		end
+	end)
+	hook.Add("HUDPaintBackground", "BlindnessDraw", function()
+		local ply = LocalPlayer()
+		if IsValid(ply) then
+			surface.SetDrawColor(255,255,255,ply:GetNWFloat("Blindness",0))

```

Comment:
This seems kind of lame, it just whites out your screen?

Can you make it draw a flare at the :ToScreen() of the laser source or something? https://pcmedia.gamespy.com/pc/image/article/118/1184679/bf3lasersight_1311840613.jpg

Reply:
maybe, I was just wanted the effect to be like a flashbang

Reply:
+ this way the target will see where the blinding laser is coming from so they can go kill the guy doing it, just making the screen white is just annoying